src/Features/Core/Portable/CodeLens/CodeLenseReferenceService.cs

```diff
@@ -167,43 +178,95 @@ private static SyntaxNode GetEnclosingCodeElementNode(Document document, SyntaxT
 
         public async Task<IEnumerable<ReferenceLocationDescriptor>> FindReferenceLocationsAsync(Solution solution, DocumentId documentId, SyntaxNode syntaxNode, CancellationToken cancellationToken)
         {
-            var document = solution.GetDocument(documentId);
-            if (document == null)
+            return await FindAsync(solution, documentId, syntaxNode,
+                async progress =>
+                {
+                    var referenceTasks = progress.Locations
+                        .Where(location => location.Kind != LocationKind.MetadataFile && location.Kind != LocationKind.None)
+                        .Distinct(LocationComparer.Instance)
+                        .Select(
+                            location =>
+                                GetDescriptorOfEnclosingSymbolAsync(solution, location, cancellationToken))
+                        .ToArray();
+
+                    await Task.WhenAll(referenceTasks).ConfigureAwait(false);
+
+                    return referenceTasks.Select(task => task.Result);
+                }, onCapped: null, searchCap: 0, cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+
+        private static SyntaxNode FindEnclosingMethodNode(Document document, int position)
+        {
+            var getSyntaxRootTask = document.GetSyntaxRootAsync();
+            var root = getSyntaxRootTask.Result;
+
+            if (!root.FullSpan.Contains(position))
             {
                 return null;
             }
 
+            var token = root.FindToken(position);
+            var node = token.Parent;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+            while (node != null)
```

Comment:
I found something like this in the find references code (GetEnclosingMethodOrPropertyOrField). It's not public and I don't need all the functionality, but it looks simpler, so I'll use similar code.

---
In reply to: [74627932](https://github.com/dotnet/roslyn/pull/13109#discussion_r74627932) [](ancestors = 74627932)